src/Controller/WishlistConfigurationAdminController.php

```diff
@@ -0,0 +1,146 @@
+            'resultHandleForm' => isset($resultHandleForm) ? $resultHandleForm : null,
+        ]);
+    }
+
+    public function statisticsAction()
+    {
+        $searchCriteria = new SearchCriteria();
+        $allTimeStatsGridFactory = $this->get('prestashop.module.blockwishlist.grid.all_time_stastistics_grid_factory');
+        $currentYearGridFactory = $this->get('prestashop.module.blockwishlist.grid.current_year_stastistics_grid_factory');
+        $currentMonthGridFactory = $this->get('prestashop.module.blockwishlist.grid.current_month_stastistics_grid_factory');
+        $currentDayGridFactory = $this->get('prestashop.module.blockwishlist.grid.current_day_stastistics_grid_factory');
+        $allTimeStatisticsGrid = $allTimeStatsGridFactory->getGrid($searchCriteria);
+        $currentYearGrid = $currentYearGridFactory->getGrid($searchCriteria);
+        $currentMonthGrid = $currentMonthGridFactory->getGrid($searchCriteria);
+        $currentDayGrid = $currentDayGridFactory->getGrid($searchCriteria);
+
+        return $this->render('@Modules/blockwishlist/views/templates/admin/statistics.html.twig', [
+            'allTimeStatisticsGrid' => $this->presentGrid($allTimeStatisticsGrid),
+            'currentYearStatisticsGrid' => $this->presentGrid($currentYearGrid),
+            'currentMonthStatisticsGrid' => $this->presentGrid($currentMonthGrid),
+            'currentDayStatisticsGrid' => $this->presentGrid($currentDayGrid),
+        ]);
+    }
+
+    public function resetStatisticsCacheAction()
+    {
+        $result = $this->cache->delete(BaseGridDataFactory::CACHE_KEY_STATS_ALL_TIME)
+            && $this->cache->delete(BaseGridDataFactory::CACHE_KEY_STATS_CURRENT_DAY)
+            && $this->cache->delete(BaseGridDataFactory::CACHE_KEY_STATS_CURRENT_MONTH)
+            && $this->cache->delete(BaseGridDataFactory::CACHE_KEY_STATS_CURRENT_YEAR);
+
+        return new JsonResponse(['success' => $result]);
+    }
+
+    /**
+     * handleForm
+     *
+     * @param array $datas
+     *
+     * @return bool
+     */
+    private function handleForm($datas)
+    {
+        $result = true;
+        if (isset($datas['WishlistPageName'])) {
+            foreach ($datas['WishlistPageName'] as $langID => $value) {
+                $result &= \Configuration::updateValue('blockwishlist_WishlistPageName', [$langID => $value]);
+            }
+        }
+
+        if (isset($datas['WishlistDefaultTitle'])) {
+            foreach ($datas['WishlistDefaultTitle'] as $langID => $value) {
+                $result &= \Configuration::updateValue('blockwishlist_WishlistDefaultTitle', [$langID => $value]);
+            }
+        }
+
+        if (isset($datas['CreateButtonLabel'])) {
+            foreach ($datas['CreateButtonLabel'] as $langID => $value) {
+                $result &= \Configuration::updateValue('blockwishlist_CreateButtonLabel', [$langID => $value]);
+            }
+        }
+
+        return $result;
```

Comment:
```suggestion
        return (bool) $result;
```
Otherwise it's an integer

Reply:
I'm in favor in modifying assignments like this: `$result = $result && [...]`